### PR TITLE
add prefetchKeys option to daml3-script submissions

### PR DESCRIPTION
**Pull request by andreaslochbihler-da**
In preparation to the newly exposed option on the LAPI. Full wiring will only happen once the Canton code drop has made into this repo.

### CODE REVIEW

File: sdk/daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/DamlScriptLtsTestRunner.scala [hunk 1]
```diff
@@ -36,12 +36,14 @@ abstract class DamlScriptLtsTestRunner extends GenericDamlScriptTestRunner {
...
       assertDamlScriptRunnerResult(
         trySubmitTestDarPath,
+        // scripts are executed in alphabetical order
         f"""DamlScriptLtsTrySubmit:authorizationError SUCCESS
            |DamlScriptLtsTrySubmit:contractKeyNotFound SUCCESS
            |DamlScriptLtsTrySubmit:contractNotActive ${expectedContractNotActiveResponse}
            |DamlScriptLtsTrySubmit:createEmptyContractKeyMaintainers SUCCESS
            |DamlScriptLtsTrySubmit:duplicateContractKey SUCCESS
            |DamlScriptLtsTrySubmit:fetchEmptyContractKeyMaintainers SUCCESS
+           |DamlScriptLtsTrySubmit:prefetchContractKeys SUCCESS
```
Comment (andreaslochbihler-da) [inline sdk/daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/DamlScriptLtsTestRunner.scala:46]: This is not really testing `trySubmit`, but this was the place where most other submission option builders where tested. LMK ff there's better place for such tests.

Comment (samuel-williams-da) [reply sdk/daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/DamlScriptLtsTestRunner.scala:46]: I think this is fine, this test has extended a bit outside of TrySubmit. You could perhaps rename the dar/module to match

Comment (andreaslochbihler-da) [reply sdk/daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/DamlScriptLtsTestRunner.scala]: I removed `Try` from the module and dar name.

**Review by samuel-williams-da — APPROVED** [review]
Just a small note relating to upgrades

File: sdk/daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/grpcLedgerClient/GrpcLedgerClient.scala [hunk 2]
```diff
@@ -457,6 +462,15 @@ class GrpcLedgerClient(
         )
     }
 
+  private def toPrefetchContractKey(key: AnyContractKey): Either[String, PrefetchContractKey] = {
+    for {
+      contractKey <- lfValueToApiValue(true, key.key.toUnnormalizedValue)
+    } yield PrefetchContractKey(
+      templateId = Some(toApiIdentifier(key.templateId)),
```
[the remark this answers is not in the document]
Comment (andreaslochbihler-da) [reply sdk/daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/grpcLedgerClient/GrpcLedgerClient.scala]: Yes, it does. Changed.

**Pull request merged**